### ISSUE TITLE
Check requests made via WP_Http for insecure resource requests

### ISSUE THREAD

**Issue by tollmanz** [tollmanz/wordpress-https-mixed-content-detector#24]
@johnbillion suggested this. I have made (https://github.com/tollmanz/wordpress-https-mixed-content-detector/pull/23) the prerequisite changes to make this much easier to save data that is passed in via the beacon. We still need to consider how to map this type of request's data on to CSP report data.

**Comment by johnbillion** [comment]
I'm thinking that rather than try to fit it into an existing policy, we add our own one (eg `http-api-src`). Plus a new violation location.

**Comment by tollmanz** [comment]
Definitely. I think the challenge is displaying it amongst other reports that have all of the usual CSP data.
